.kokoro/dependencies.sh

```diff
@@ -24,4 +24,4 @@ echo $JOB_TYPE
 export MAVEN_OPTS="-Xmx1024m -XX:MaxPermSize=128m"
 
 mvn install -DskipTests=true -B -V
-mvn -B dependency:analyze -DfailOnWarning=true
+mvn -B dependency:analyze -DfailOnWarning=true -DignoreNonCompile
```

Comment:
why DignoreNonCompile?

Reply:
dependencies such as auto-value were flagging as unused without this (this forces the analyzer to ignore deps with provided scope)

Reply:
For `auto-value`, we should actually move it to the annotation processor path (rather than a provided scope dependency). Example: https://github.com/googleapis/google-auth-library-java/blob/master/pom.xml#L213-L230

`auto-value-annotations` still needs to be a compile dependency.

Reply:
Thanks @chingor13! I made that change, but looks like we still need it for 
``` 
  <dependency>
      <groupId>io.grpc</groupId>
      <artifactId>grpc-grpclb</artifactId>
      <scope>runtime</scope>
    </dependency>
```

Reply:
Ok, that makes sense then. In other places, we've explicitly excluded the dependency if we know it's not referenced at compile time. Example: https://github.com/googleapis/java-speech/blob/master/pom.xml#L280-L292

Reply:
I think that's preferable: instead of -DignoreNonCompile, we should add the following to google-cloud-bigtable/pom.xml

```xml
   <pluginManagement>
          <plugins>
            <plugin>
              <groupId>org.apache.maven.plugins</groupId>
              <artifactId>maven-dependency-plugin</artifactId>
              <configuration>
                <ignoredUnusedDeclaredDependencies>io.grpc:grpc-grpclb:jar:1.24.0</ignoredUnusedDeclaredDependencies>
              </configuration>
            </plugin>
          </plugins>
        </pluginManagement>
```

Reply:
Or better yet:

```
<pluginManagement>
          <plugins>
            <plugin>
              <groupId>org.apache.maven.plugins</groupId>
              <artifactId>maven-dependency-plugin</artifactId>
              <configuration>
                <usedDependencies>io.grpc:grpc-grpclb</usedDependencies>
                <usedDependencies>io.grpc:grpc-auth</usedDependencies>
              </configuration>
            </plugin>
          </plugins>
        </pluginManagement>
```